hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/gcs/MessageBatch.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.utilities.sources.helpers.gcs;
+
+import java.util.List;
+
+/**
+ * A batch of messages fetched from Google Cloud Pubsub within the metadata fetcher of
+ * Incremental GCS ingestion module.
+ */
+public class MessageBatch {
```

Comment:
Why is this only needed for GCS and not for S3?

Reply:
Conceptually some equivalent work happens on the S3 side of things using filteredEventRecords - https://github.com/apache/hudi/blob/master/hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/S3EventsMetaSelector.java#L147

It's just that I've created separate classes and functions for some of that logic. The actual code inside is very tied to the impl. Broadly speaking, I didn't aim to refactor the S3 code when implementing this piece. /cc @codope for visibility and any thoughts.

Reply:
I suggest to use this in S3 as well. But it can be taken up as a follow up

Reply:
This can be taken up as part of [HUDI-4929](https://issues.apache.org/jira/browse/HUDI-4929) - Refactor code that is common to all ingestions from cloud sources.